### ISSUE TITLE
How to disable auto completion pop up when dot is typed?

### ISSUE THREAD

**Issue by evandrocoan** [tomv564/LSP#685]
Related:
1. https://github.com/tomv564/LSP/issues/203 Spaces don't trigger autocompletion

I see the related code here:
```python
File: sublime-text-3/Packages/LSP/plugin/completion.py
168:     def register_trigger_chars(self, session: Session) -> None:
169:         completion_triggers = self.view.settings().get('auto_complete_triggers', [])
170:         view_language = self._view_language(session.config.name)
171:         if view_language:
172:             for language in session.config.languages:
173:                 if language.id == view_language:
174:                     for scope in language.scopes:
175:                         # log(2, "registering", self.trigger_chars, "for", scope)
176:                         scope_trigger = next(
177:                             (trigger for trigger in completion_triggers if trigger.get('selector', None) == scope),
178:                             None
179:                         )
180:                         if scope_trigger:
181:                             scope_trigger['characters'] = "".join(self.trigger_chars)
182:                         else:
183:                             completion_triggers.append({
184:                                 'characters': "".join(self.trigger_chars),
185:                                 'selector': scope
186:                             })
187: 
```

**Comment by tomv564** [comment]
Is there something incorrect about the autocomplete behaviour, or is it a preference to not see it?

**Comment by evandrocoan** [comment]
I as I set to commit auto-completion with the `Enter` key, it just became very annoying recently because most times I had to close it with `Escape`. It is preferable to press `Ctrl+Space` to open the auto-completion if I need it after a dot.

**Comment by tomv564** [comment]
To avoid adding yet another setting to LSP, I wonder if we can make LSP respect your Sublime configuration here. Would it work if LSP only registered the trigger characters if you had nothing configured for a given scope?

**Comment by evandrocoan** [comment]
What do you mean by scope? Is it this configuration?
```js
  "default_clients":
  {
    "pyls":
    {
      "command": ["pyls"],
      "scopes": ["source.python"],
      "syntaxes": ["Packages/Python/Python.sublime-syntax"],
      "languageId": "python"
    },
```

I did not configure the `"scopes": ["source.python"]` on my user settings.

**Comment by tomv564** [comment]
Sorry, that was super unclear. 
So I think LSP is aggressive in modifying the user's Sublime `auto_complete_triggers`:

```jsonc
    "auto_complete_triggers": [ {"selector": "text.html", "characters": "<"} ],
```

My proposal is to disable the registering of additional auto complete triggers in LSP if you add `{ "selector": "source.python", "characters": "" }`

What do you think?

**Comment by evandrocoan** [comment]
If I can add `{ "selector": "source.python", "characters": "" }` into the `LSP` settings, for the `pyls` plugin, then it is fine. I would not like to add it to my main/global `Preferences.sublime-settings` because I would overwrite all packages settings for this setting.

**Comment by jfcherng** [comment]
@evandrocoan syntax-specific settings may be an alternative?
![image](https://user-images.githubusercontent.com/6594915/63443812-0cafc700-c468-11e9-8b6b-747bfceba698.png)

**Comment by evandrocoan** [comment]
I do not like that neither. They are also global (per language) settings configuration.

What is the problem with adding a LSP setting to the `LSP.sublime-settings`?

**Comment by tomv564** [comment]
I understand you think it's LSP's problem and that's why the setting should live there. I don't want to add new settings to LSP for every single exception, so my offer is to respect the user's Sublime configuration.

**Comment by evandrocoan** [comment]
I just managed to add the setting. I just cannot believe how hard it was.

Now I understand why you do not want to add new settings. It is incredibly hard. Any change involves a lot of refactoring by changing several function calls signatures.

I managed to get out with less by adding it in some place I did not liked to add it, but involved `0` refactoring. 

Ideally, I would add this setting for each language the language server provides, but to avoid any refactoring to the internal settings building I added it globally to all languages a language server provides.

Why it is not difficult to add new settings? It is because language servers are insane to satisfy?
